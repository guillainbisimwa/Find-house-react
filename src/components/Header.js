import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FaUserNinja, FaPlayCircle } from 'react-icons/fa';
import userActions from '../redux/actions/UserActions';
import history from '../helpers/History';

const Header = ({ user }) => {
  const dispatch = useDispatch();

  const logOut = () => {
    history.push('/login');
    dispatch(userActions.logout());
  };

  return (
    <div className="bg-login-bg">
      <div className="bg-black bg-opacity-50 h-auto p-5 text-white">
        <header className="flex mb-10 justify-between">
          <Link to="/">
            <img alt="logo" src="https://github.com/guillainbisimwa/Find-house-api-doc/raw/main/source/images/logo.png" width="150" />
          </Link>
          <div className="mt-5 flex gap-4 md:mx-10 items-center">
            <FaUserNinja />
            <Link to="/">{user.name}</Link>
            <Link to="/favorites">My favorite</Link>
            <Link to="/login" onClick={logOut} className="bg-black bg-opacity-40 py-2 px-6 text-white font-semibold rounded-full shadow-2xl focus:outline-none">Logout</Link>
          </div>
        </header>
        <div className="pt-10 mt-10 w-4/4 sm:w-3/4 md:w-2/4 xl:w-2/4">
          <h2 className="intro-x font-bold text-white text-3xl xl:text-5xl text-left pb-2"> Find your house anywhere</h2>
          <p className="">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit,
            sed do eiusmod tempor incididunt ut labore et dolore
            magna aliqua. Ut enim ad minim veniam,
          </p>
          <div className="flex gap-4 my-4">
            <button type="button" className="bg-primary py-3 mt-3 px-6 text-white block font-semibold rounded-full shadow-2xl focus:outline-none">
              Start free
            </button>
            <div className="flex gap-4 mt-4 items-center">
              <FaPlayCircle />
              <div>Free video</div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

Header.propTypes = {
  user: PropTypes.oneOfType([PropTypes.object]).isRequired,
};

export default Header;
