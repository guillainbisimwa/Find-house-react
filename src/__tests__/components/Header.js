import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import Header from '../../components/Header';
import store from '../../redux/reducers';

describe('Header component', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Header user={{ name: 'test' }} />
        </MemoryRouter>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
