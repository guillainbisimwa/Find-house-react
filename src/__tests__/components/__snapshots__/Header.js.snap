// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<div
  className="bg-login-bg"
>
  <div
    className="bg-black bg-opacity-50 h-auto p-5 text-white"
  >
    <header
      className="flex mb-10 justify-between"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          src="https://github.com/guillainbisimwa/Find-house-api-doc/raw/main/source/images/logo.png"
          width="150"
        />
      </a>
      <div
        className="mt-5 flex gap-4 md:mx-10 items-center"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M325.4 289.2L224 390.6 122.6 289.2C54 295.3 0 352.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-70.2-54-127.1-122.6-133.2zM32 192c27.3 0 51.8-11.5 69.2-29.7 15.1 53.9 64 93.7 122.8 93.7 70.7 0 128-57.3 128-128S294.7 0 224 0c-50.4 0-93.6 29.4-114.5 71.8C92.1 47.8 64 32 32 32c0 33.4 17.1 62.8 43.1 80-26 17.2-43.1 46.6-43.1 80zm144-96h96c17.7 0 32 14.3 32 32H144c0-17.7 14.3-32 32-32z"
          />
        </svg>
        <a
          href="/"
          onClick={[Function]}
        >
          test
        </a>
        <a
          href="/favorites"
          onClick={[Function]}
        >
          My favorite
        </a>
        <a
          className="bg-black bg-opacity-40 py-2 px-6 text-white font-semibold rounded-full shadow-2xl focus:outline-none"
          href="/login"
          onClick={[Function]}
        >
          Logout
        </a>
      </div>
    </header>
    <div
      className="pt-10 mt-10 w-4/4 sm:w-3/4 md:w-2/4 xl:w-2/4"
    >
      <h2
        className="intro-x font-bold text-white text-3xl xl:text-5xl text-left pb-2"
      >
         Find your house anywhere
      </h2>
      <p
        className=""
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
      </p>
      <div
        className="flex gap-4 my-4"
      >
        <button
          className="bg-primary py-3 mt-3 px-6 text-white block font-semibold rounded-full shadow-2xl focus:outline-none"
          type="button"
        >
          Start free
        </button>
        <div
          className="flex gap-4 mt-4 items-center"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm115.7 272l-176 101c-15.8 8.8-35.7-2.5-35.7-21V152c0-18.4 19.8-29.8 35.7-21l176 107c16.4 9.2 16.4 32.9 0 42z"
            />
          </svg>
          <div>
            Free video
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
